staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -158,6 +159,18 @@ func (a *mutatingDispatcher) callAttrMutatingHook(ctx context.Context, h *v1beta
 		return apierrors.NewInternalError(err)
 	}
 
+	// If object is mutated, record the mutation in audit log
+	if !bytes.Equal(objJS, patchedJS) {
+		// use a key prefix different from the webhook name to avoid key conflict, where:
+		//     "webhooktrace" is the name of the virtual plugin,
+		//     "admission.k8s.io" is the name of the org,
+		//     webhook name (h.Name) is the key name,
+		//     "mutation" is the value
+		if err := attr.AddAnnotation("webhooktrace.admission.k8s.io/"+h.Name, "mutation"); err != nil {
```

Comment:
Should we distinguish between metalevel auditing and object level? In the latter case we might even want to see the actual patch.

Reply:
would prefer `mutation.webhook..admission.k8s.io/name: true/false`.

Reply:
> Should we distinguish between metalevel auditing and object level? In the latter case we might even want to see the actual patch.

good point. I'm looking at if that's achievable here

> Is this blocked on anything?

need to rebase on reinvocation PR https://github.com/kubernetes/kubernetes/pull/78080 to treat the second pass specially. According to `AddAnnotation` doc, overwriting existing key with new value results in an error

Reply:
> Should we distinguish between metalevel auditing and object level? In the latter case we might even want to see the actual patch.

done. Adding/updating integration/e2e tests to verify behavior under different audit levels